obspy/taup/tau.py

```diff
@@ -663,3 +728,174 @@ def create_taup_model(model_name, output_dir, input_dir):
+        :param plot_type: type of plot to create. Defaults to ``spherical``,
+            but can also be ``cartesian``.
+        :type plot_type: str
+        :param phases: List of phase names which should be used within the
+            plot. Defaults to ``['P', 'S']``.
+        :type phases: list of str
+        :param model: string containing the model to use. Defaults to 'iasp91'.
+        :type model: str
+        :param plot_all: By default all rays, even those travelling in the
+            other direction and thus arriving at a distance of *360 - x*
+            degrees are shown. Set this to ``False`` to only show rays
+            arriving at exactly *x* degrees.
+        :type plot_all: bool
+        :param legend: If boolean, specify whether or not to show the legend
+            (at the default location.) If a str, specify the location of the
+            legend. If you are plotting a single phase, you may consider using
+            the ``label_arrivals`` argument.
+        :type legend: bool or str
+        :param label_arrivals: Label the arrivals with their respective phase
+            names. This setting is only useful if you are plotting a single
+            phase as otherwise the names could be large and possibly overlap
+            or clip. Consider using the ``legend`` parameter instead if you
+            are plotting multiple phases.
+        :type label_arrivals: bool
+        :param fig: Figure to plot into. If not given, a new figure instance
+            will be created.
+        :type fig: :class:`matplotlib.axes.Axes
+        :param ax: Axes to plot to. If not given, a new figure with an axes
+            will be created.
+        :type ax: :class:`matplotlib.axes.Axes`
+
+        :returns: ax
+        :rtype: :class:`matplotlib.axes.Axes`
+
+        .. rubric:: Example
+
+        >>> from obspy.taup import raypath_plot
+        >>> import matplotlib.pyplot as plt
+        >>> fig, ax = plt.subplot(111, polar=True)
+        >>> ax = raypath_plot(source_depth=100, ax=ax, fig=fig)
+        >>> plt.show()
+
+        .. plot::
+
+        from obspy.taup.tau import raypath_plot
+        import matplotlib.pyplot as plt
+
+        fig, ax = plt.subplot(111, polar=True)
+
+        ax = raypath_plot(source_depth=100, ax=ax, fig=fig)
+
+        plt.show()
+
+        """
+
+        plotted = False  # a flag to check if any arrival exists
+
+        # compute the requested arrivals:
+        model = TauPyModel(model)
+
+        # calculate the arrival times and plot vs. epicentral distance:
+        degrees = np.linspace(min_degree, max_degree, nrecs)
+        for degree in degrees:
+            try:
+                arrivals = model.get_ray_paths(source_depth, degree,
+                                               phase_list=phases)
+                arrivals.plot(show=False, ax=ax, plot_type=plot_type,
+                              plot_all=plot_all,
+                              legend=legend)
```

Comment:
This might also have the problem of phases changing colors - I think there should be way to resolve this although it might not be trivial here.